src/utilities/error.c

```diff
@@ -252,11 +268,6 @@ HYPRE_GetErrorMessages(char **buffer, HYPRE_Int *bufsz)
    *buffer = hypre_CTAlloc(char, *bufsz, HYPRE_MEMORY_HOST);
    hypre_TMemcpy(*buffer, err.memory, char, *bufsz, HYPRE_MEMORY_HOST, HYPRE_MEMORY_HOST);
 
-   hypre_TFree(err.memory, HYPRE_MEMORY_HOST);
```

Comment:
We should call `hypre_error_clear_messages` here.  Messages are "handled" by either calling `Get` or calling `Print`, and once that's done they should be cleared.

Reply:
Sounds good! But the call on `HYPRE_Finalize` would be redundant in this case. Is this fine?

Reply:
Yes, that's totally fine.  The `TFree` will do nothing since the pointer will be `NULL` the second time.